File: app/src/main/java/huce/fit/appreadstories/story/list/download/StoryDownloadImpl.java
package huce.fit.appreadstories.story.list.download;

import android.content.Context;

import huce.fit.appreadstories.sqlite.AppDao;
import huce.fit.appreadstories.sqlite.AppDatabase;
import huce.fit.appreadstories.story.list.BaseStoryListImpl;

public class StoryDownloadImpl extends BaseStoryListImpl implements StoryDownloadPresenter {

    private static final String TAG = "StoryDownloadImpl";
    private final StoryDownloadView mStoryDownloadView;
    private final AppDao mAppDao;

    StoryDownloadImpl(StoryDownloadView storyDownloadView, Context context) {
        super(context);
        mStoryDownloadView = storyDownloadView;
        mAppDao = AppDatabase.getInstance(context).appDao();
    }

    @Override
    public void getData() {
        mStoryDownloadView.setData(mAppDao.getAllStory());
    }
}

File: app/src/main/java/huce/fit/appreadstories/story/list/download/StoryDownloadImpl.kt
package huce.fit.appreadstories.story.list.download

import android.content.Context
import huce.fit.appreadstories.sqlite.AppDatabase
import huce.fit.appreadstories.story.list.BaseStoryListImpl

class StoryDownloadImpl(private val storyDownloadView: StoryDownloadView, context: Context) :
    BaseStoryListImpl(context), StoryDownloadPresenter {

    companion object {
        private const val TAG = "StoryDownloadImpl"
    }

    private var mAppDao = AppDatabase.getInstance(context).appDao()

    override fun getData() {
        storyDownloadView.setData(mAppDao.allStory)
    }
}
